python/ray/dashboard/client/src/contexts/ThemeContext.tsx

```diff
@@ -0,0 +1,62 @@
+import React, {
+  createContext,
+  ReactNode,
+  useContext,
+  useEffect,
+  useState,
+} from "react";
+
+type ThemeMode = "light" | "dark";
+
+type ThemeContextType = {
+  mode: ThemeMode;
+  toggleTheme: () => void;
+};
+
+const ThemeContext = createContext<ThemeContextType | undefined>(undefined);
+
+type ThemeProviderProps = {
+  children: ReactNode;
+};
+
+export const ThemeProvider: React.FC<ThemeProviderProps> = ({ children }) => {
```

Comment:
let's not name this `ThemeProvider` since it can be confused with an MUI ThemeProvider class.

How about `ThemeSettingsProvider`?

And let's just rename `Theme` to `ThemeSettings` in this file in general.